lib/ansible/plugins/inventory/aws_ssm.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: aws_ssm
+    plugin_type: inventory
+    author:
+      - Andrew Hibbert <andrew_hibbert@hotmail.com>
+    requirements:
+    - boto3
+    - botocore
+    short_description: AWS SSM inventory source
+    description:
+      - Fetch parameters via path
+      - Add groups, hosts and variables based on path
+    options:
+      plugin:
+        description: token that ensures this is a source file for the 'aws_ssm' plugin.
+        required: True
+        choices: ['aws_ssm']
+      path:
+        description: Path to search for in ssm
+        required: True
+      region:
+        description: Region to retrieve SSM parameter from
+        required: True
+      decrypt:
+        description: Whether to decrypt secrets or not
+        required: False
+        default: True
+'''
+
+EXAMPLES = '''
+plugin: aws_ssm
+path: prod
+region: us-east-1
+aws_profile: default
+'''
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.utils.display import Display
+
+try:
+    import boto3
+    import botocore
+except ImportError:
+    raise AnsibleError("The ssm dynamic inventory plugin requires boto3 and "
+                       "botocore")
+
+
+display = Display()
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'aws_ssm'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+    def verify_file(self, path):
+        ''' Verify plugin configuration file. '''
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('aws_ssm.yml', 'aws_ssm.yaml')):
+                return True
+        display.debug("aws_ssm inventory filename must end with 'aws_ssm.yml' "
+                      "or 'aws_ssm.yaml'")
+        return False
+
+    def parse(self, inventory, loader, path, cache=True):
+        ''' Retrieve data '''
+        super(InventoryModule, self).parse(inventory, loader, path)
+        cache_key = self._get_cache_prefix(path)
+        config_data = self._read_config_data(path)
```

Comment:
`_read_config_data` reads and validates the options in the config file, so you should use self.get_option(option_name) instead of config_data.get(option_name) in this plugin to make sure option defaults and other sources (like environment variables) are used.

Reply:
I have made this change